db/db_impl/db_impl_write.cc

```diff
@@ -1473,6 +1481,28 @@ void DBImpl::MaybeFlushStatsCF(autovector<ColumnFamilyData*>* cfds) {
   }
 }
 
+Status DBImpl::TrimMemtableHistory() {
+  autovector<ColumnFamilyData*> cfds;
+  ColumnFamilyData* tmp_cfd;
+  while ((tmp_cfd = trim_history_scheduler_.TakeNextColumnFamily()) !=
+         nullptr) {
+    cfds.push_back(tmp_cfd);
+  }
+  for (auto& cfd : cfds) {
+    autovector<MemTable*> to_delete;
+    cfd->imm()->TrimHistory(&to_delete, cfd->mem()->ApproximateMemoryUsage());
+    cfd->imm()->ResetTrimHistoryNeeded();
```

Comment:
I believe this needs to be reset every time a new version is installed, not just here. Otherwise, flush will cause this to be false and it will never be reset to true.

Reply:
Right now `CheckMemtableFull()` sets the flag to true if total memory usage exceeds the limit, `DBImpl::TrimMemtableHistory()` (called by `PreprocessWrite`) calls `TrimHistory` and resets the flag to false. Are you suggesting we should also reset the flag to false when `MemTableList::TrimHistory` is called elsewhere?